src/pykeen/nn/node_piece/anchor_search.py

```diff
@@ -222,6 +222,152 @@ def __call__(self, edge_index: numpy.ndarray, anchors: numpy.ndarray, k: int) ->
         return self.select(pool=pool, k=k)
 
 
+class SparseBFSSearcher(ScipySparseAnchorSearcher):
+    """Find closest anchors using :mod:`torch_sparse` on a GPU."""
+
+
+    @staticmethod
+    def create_adjacency(
+        edge_index: numpy.ndarray,
+    ) -> torch.tensor:
+        """
+        Create a sparse adjacency matrix (in the form of the edge list) from a given edge index.
+
+        :param edge_index: shape: (2, m)
+            the edge index
+
+        :return: shape: (2, 2m + n)
+            edge list with inverse edges and self-loops
+        """
+
+        # infer shape
+        num_entities = edge_index.max().item() + 1
+        edge_index = torch.tensor(edge_index, dtype=torch.long)
+
+        # symmetric + self-loops
+        # TODO what if the edge index already has inverse edges?
```

Comment:
maybe remove them?
```suggestion
edge_index = edge_index.unique(dim=0)
```

Reply:
I think I postet that in the wrong order 😅 

What if you first add the new edges (irrespective of whether they may already be present), and *afterwards* perform unique?

Reply:
If we work with `bool` tensor / masks, the multiplicity of edges would not result in any change.

Reply:
alright, made deduplication in https://github.com/pykeen/pykeen/pull/990/commits/ee3995d481b2f1d1c3a63c99e2f01f1e35c33fce